src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerEndpoint.java

```diff
@@ -166,15 +179,51 @@ public boolean equals(Object obj) {
         }
         return true;
     }
-    
+
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
     @Extension
     public static class DescriptorImpl extends Descriptor<DockerServerEndpoint> {
+
+        private String defaultHost;
+
+        public String getDefaultHost() {
+            return defaultHost;
+        }
+
+        public DescriptorImpl() {
+            load();
+        }
+
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+            defaultHost = json.getString("defaultHost");
+            save();
+            return true;
+        }
+
+        public String getDefaultDockerHost() {
+            if (StringUtils.isNotBlank(defaultHost))
+                return defaultHost;
+            String env = System.getenv("DOCKER_HOST");
```

Comment:
This is an environment variable on master and so is worse than useless as a default. :bug: